core/trino-main/src/main/java/io/trino/util/ColumnDefaultOptions.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.security.AccessControl;
+import io.trino.spi.type.Type;
+import io.trino.sql.PlannerContext;
+import io.trino.sql.analyzer.ExpressionAnalyzer;
+import io.trino.sql.analyzer.Scope;
+import io.trino.sql.ir.optimizer.IrExpressionEvaluator;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.NodeRef;
+import io.trino.sql.tree.Parameter;
+
+import java.util.Map;
+
+import static io.trino.spi.StandardErrorCode.INVALID_LITERAL;
+import static io.trino.sql.analyzer.ConstantEvaluator.evaluateConstant;
+import static io.trino.sql.analyzer.ExpressionAnalyzer.createConstantAnalyzer;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static io.trino.sql.planner.LogicalPlanner.noTruncationCast;
+import static java.util.Locale.ENGLISH;
+
+public final class ColumnDefaultOptions
```

Comment:
We don't need a generic utility class for this. Move the method to ExpressionAnalyzer (see other methods such as `analyzeJsonPathInvocation`, `analyzePatternRecognitionExpression`, etc.)

Reply:
Moved the method to `ExpressionAnalyzer`.